packages/@tailwindcss-cli/src/commands/build/index.ts

```diff
@@ -207,15 +208,28 @@ export async function handle(args: Result<ReturnType<typeof options>>) {
             args['--input'] ?? base,
           )
 
-          build = compile(input).build
-          compiledCss = build(candidates)
+          // Create a new compiler, given the new `input`
+          compiler = compile(input)
+
+          // Re-scan the directory to get the new `candidates`
+          scanDirResult = scanDir({ base, contentPaths: compiler.globs })
```

Comment:
Should we make `contentPaths` a `Vec<GlobEntry>` or something like that and do any merging / concatenation on the Rust side? I think it'd be ideal that the base and glob are separate for as long as possible.

Reply:
Yep, we ended up doing exactly this, because we ran into an issue: https://github.com/tailwindlabs/tailwindcss/pull/14067#discussion_r1695607073

Essentially `base` was referring to the `process.cwd()` (for auto content detection. But globs were relative to the base path of the `input` file, which could live in `process.cwd() + './src/styles'` for example.

`Vec<GlobEntry>`, makes a lot of sense here thanks!